qiskit/optimization/algorithms/warm_start_qaoa_optimizer.py

```diff
@@ -0,0 +1,258 @@
+
+        results = []  # type: List[MinimumEigenOptimizationResult]
+        for relaxed_solution, value in relaxed_solutions:
+            # Set the solver using the result of the pre-solver.
+            initial_variables = self._create_initial_variables(relaxed_solution)
+            # todo: no such setter
+            self._qaoa.initial_state = self._create_initial_state(initial_variables)
+            # todo: no such setter
+            self._qaoa.mixer = self._create_mixer(initial_variables)
+
+            # approximate ground state of operator using min eigen solver.
+            results.append(self._solve_internal(operator, offset, qubo_problem, problem))
+
+        # todo: handle converters and _interpret
+        if len(results) == 1:
+            return results[0]
+        else:
+            return self._aggregator.aggregate(results, qubo_problem, None)
+
+    def _create_initial_variables(self, solution: List[float]) -> List[float]:
+        """
+        Creates initial variable values to warm start QAOA.
+
+        Args:
+            solution: a solution obtained for the relaxed problem.
+
+        Returns:
+            A list of initial variables constructed from a relaxed solution.
+        """
+        initial_variables = []
+
+        for variable in solution:
+            if variable < self._epsilon:
+                initial_variables.append(self._epsilon)
+            elif variable > 1. - self._epsilon:
+                initial_variables.append(1. - self._epsilon)
+            else:
+                initial_variables.append(variable)
+
+        return initial_variables
+
+    @staticmethod
+    def _create_initial_state(initial_variables: List[float]) -> QuantumCircuit:
+        """
+        Creates an initial state quantum circuit to warm start QAOA.
+
+        Args:
+            initial_variables: Already created initial variables.
+
+        Returns:
+            A quantum circuit that represents initial state.
+        """
+        circuit = QuantumCircuit(len(initial_variables))
+
+        for index, relaxed_value in enumerate(initial_variables):
+            theta = 2 * np.arcsin(np.sqrt(relaxed_value))
+            circuit.ry(theta, index)
+
+        return circuit
+
+    @staticmethod
+    def _create_mixer(initial_variables: List[float]) -> QuantumCircuit:
+        """
+        Creates the evolved mixer circuit as Ry(theta)Rz(-2beta)Ry(-theta).
+
+        Args:
+            initial_variables: Already created initial variables.
+
+        Returns:
+            A quantum circuit to be used as a mixer in QAOA.
+        """
+        circuit = QuantumCircuit(len(initial_variables))
+        beta = Parameter("beta")
+
+        for index, relaxed_value in enumerate(initial_variables):
+            theta = 2 * np.arcsin(np.sqrt(relaxed_value))
+
+            circuit.ry(theta, index)
+            circuit.rz(-2.0 * beta, index)
+            circuit.ry(-theta, index)
```

Comment:
```suggestion
            circuit.ry(-theta, index)
            circuit.rz(-2.0 * beta, index)
            circuit.ry(theta, index)
```
The minus angle should be first. See fig 1 here: https://arxiv.org/pdf/2009.10095.pdf